http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskServiceInterp.scala

```diff
@@ -90,6 +101,22 @@ class DiskServiceInterp[F[_]: Parallel](config: PersistentDiskConfig,
       }
     } yield ()
 
+  override def lookupSourceDiskLink(userInfo: UserInfo, ctx: AppContext)(
+    sourceDiskReq: SourceDiskRequest
+  )(implicit as: Ask[F, AppContext]): F[DiskLink] =
+    for {
+      sourceDisk <- getDisk(userInfo, CloudContext.Gcp(sourceDiskReq.googleProject), sourceDiskReq.name).recoverWith {
+        case _: DiskNotFoundException =>
+          F.raiseError(BadRequestException("source disk does not exist", Option(ctx.traceId)))
+      }
+      maybeGoogleDisk <- googleDiskService.getDisk(sourceDiskReq.googleProject, sourceDisk.zone, sourceDisk.name)
```

Comment:
Note that this is actually verifying the disk exists in googleland. This is not strictly necessary. My main desire is to be able to call `googleDisk.getSelfLink` below instead of constructing the link via string concatenation.